pkgpanda/build/__init__.py

```diff
@@ -1177,24 +1218,35 @@ def clean():
 
     # TOOD(cmaloney): Disallow writing to well known files and directories?
     # Source we checked out
-    cmd.volumes.update({
-        # TODO(cmaloney): src should be read only...
-        cache_abs("src"): "/pkg/src:rw",
-        # The build script
-        build_script: "/pkg/build:ro",
-        # Getting the result out
-        cache_abs("result"): "/opt/mesosphere/packages/{}:rw".format(pkg_id),
-        install_dir: "/opt/mesosphere:ro"
-    })
-
+    if is_windows:
+        cmd.volumes.update({
+            # Source directory
+            cache_abs("src"): PKG_DIR + "/src:rw",
+            # The build script
+            # 2DO: we cannot pass a file to a volume mount on windows, only directory
+            package_dir: PKG_DIR + "/build:ro",
+            # Getting the result out
+            cache_abs("result"): install_root + "/" + PACKAGES_DIR + "/{}:rw".format(pkg_id),
+            # 2DO: windows docker does not suport overlapping mounts so push into a temporary directory in case needed
+            install_dir: install_root + "/install_dir:ro"
+        })
+    else:
+        cmd.volumes.update({
+            # TODO(cmaloney): src should be read only...
+            cache_abs("src"): PKG_DIR + "/src:rw",
+            # The build script
+            build_script: PKG_DIR + "/build:ro",
+            # Getting the result out
+            cache_abs("result"): install_root + "/" + PACKAGES_DIR + "/{}:rw".format(pkg_id),
+            install_dir: install_root + ":ro"
+        })
```

Comment:
This is duplicating a lot. We should just change the Dockerfile so that it expects a directory, since that's what works on Windows, and remove the `if is_windows:` branch.

Reply:
Yep. The install_dir for now is different due to a windows/docker-on-windows limitation. They will be fixing this, but for now we have a difference. Once this pull request is completed I will be trying to move to a newer operating system to see if we can revert the windows specific change here.

Reply:
These lines are identical in both blocks, and should not need to be duplicated:
```
cache_abs("src"): PKG_DIR + "/src:rw",
cache_abs("result"): install_root + "/" + PACKAGES_DIR + "/{}:rw".format(pkg_id),
```

Reply:
I have removed the common code into a single block.